### PR TITLE
LPS-147111 | LPS-147112

### PR DESCRIPTION
**Pull request by carolmariaabb**
Related Issue:
https://issues.liferay.com/browse/LPS-147111
https://issues.liferay.com/browse/LPS-147112

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:relevant

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/AttachmentObjectFieldBusinessType.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Carolina Barbosa
+ */
+@Component(
+	immediate = true,
+	property = "object.field.business.type.key=" + ObjectFieldConstants.BUSINESS_TYPE_ATTACHMENT,
+	service = {
+		AttachmentObjectFieldBusinessType.class, ObjectFieldBusinessType.class
```
Comment (rodrigopaulino) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/AttachmentObjectFieldBusinessType.java:34]: Do you think we will reference the `AttachmentObjectFieldBusinessType` service anywhere directly in the future?

File: modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/object/dynamic/data/mapping/form/field/type/internal/attachment/AttachmentDDMFormFieldType.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Carolina Barbosa
+ */
+@Component(
+	immediate = true,
+	property = {
+		"ddm.form.field.type.name=" + ObjectDDMFormFieldTypeConstants.ATTACHMENT,
+		"ddm.form.field.type.system=true"
```
Comment (rodrigopaulino) [inline modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/object/dynamic/data/mapping/form/field/type/internal/attachment/AttachmentDDMFormFieldType.java:33]: Is this really used anywhere?

**Review by rodrigopaulino — COMMENTED** [review]
Hey, @carolmariaabb. Nice catch on moving the `ObjectRelationship` field to their respective modules. It makes me happy that we're freeing DDM from app-specific code/customization. 👏🏻 👏🏻 😄 

I've made some comments that aren't at all impediments but I think it's worth to be mindful about them so we don't propagate unnecessary practices. 😉 

Once again. Kudos! 🎉

File: modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/object/dynamic/data/mapping/form/field/type/internal/attachment/AttachmentDDMFormFieldTemplateContextContributor.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Carolina Barbosa
+ */
+@Component(
+	immediate = true,
+	property = "ddm.form.field.type.name=" + ObjectDDMFormFieldTypeConstants.ATTACHMENT,
+	service = {
+		AttachmentDDMFormFieldTemplateContextContributor.class,
```
[the remark this answers is not in the document]
Comment (carolmariaabb) [reply modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/object/dynamic/data/mapping/form/field/type/internal/attachment/AttachmentDDMFormFieldTemplateContextContributor.java:34]: I do not think so. I will remove it, thanks!

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/configuration/activator/FFBusinessTypeAttachmentConfigurationActivator.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+ * @author Carolina Barbosa
+ */
+@Component(
+	configurationPid = "com.liferay.object.web.internal.configuration.FFBusinessTypeAttachmentConfiguration",
+	immediate = true,
+	service = FFBusinessTypeAttachmentConfigurationActivator.class
+)
+public class FFBusinessTypeAttachmentConfigurationActivator {
```
[the remark this answers is not in the document]
Comment (carolmariaabb) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/configuration/activator/FFBusinessTypeAttachmentConfigurationActivator.java:34]: You're right! Thanks for the tip!

**Comment by carolmariaabb** [comment]
ci:test:sf

File: modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/object/dynamic/data/mapping/form/field/type/internal/attachment/AttachmentDDMFormFieldType.java (hunk 2 above)
Comment (rodrigopaulino) [reply modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/object/dynamic/data/mapping/form/field/type/internal/attachment/AttachmentDDMFormFieldType.java:33]: Oh, ok. No problem. Thanks for letting me know, @carolmariaabb

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:reevaluate:1317401_2165

**Comment by liferay-forms** [comment]
Hey @carolmariaabb, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @carolmariaabb, can you please check if these failures are related to your changes? Thanks

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**